src/query_parser/v3/lexer.rl

```diff
@@ -0,0 +1,644 @@
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  inf => {
+    tok.pos = ts-q->raw;
+    tok.s = ts;
+    tok.len = te-ts;
+    tok.numval = *ts == '-' ? -INFINITY : INFINITY;
+    RSQuery_Parse_v3(pParser, NUMBER, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  quote => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, QUOTE, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  or => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, OR, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  lp => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, LP, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  rp => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, RP, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  lb => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, LB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  rb => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, RB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  colon => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, COLON, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  semicolon => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, SEMICOLON, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  minus =>  {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, MINUS, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  tilde => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, TILDE, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  star => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, STAR, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+   percent => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, PERCENT, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  lsqb => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, LSQB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  rsqb => {
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, RSQB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+  space;
+  punct;
+  cntrl;
+
+  escaped_term => {
+    tok.len = te-ts;
+    tok.s = ts;
+    tok.numval = 0;
+    tok.pos = ts-q->raw;
+    RSQuery_Parse_v3(pParser, TERM, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  unescaped_tag => {
+    tok.numval = 0;
+    tok.len = 1;
+    tok.s = ts;
+    RSQuery_Parse_v3(pParser, COLON, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+
+    tok.s = ts + 1;
+    tok.pos = tok.s - q->raw;
+    RSQuery_Parse_v3(pParser, LB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+
+    tok.len = te - (ts + 3);
+    tok.s = ts + 2;
+
+    if(tok.s[0] == 'w' && tok.s[1] == '\'' && tok.s[tok.len-1] == '\'') {
```

Comment:
Are we sure token is not too short?

Reply:
New validation and more tests included in [627bd28](https://github.com/RediSearch/RediSearch/pull/4433/commits/627bd28ef70f98bc87f8dc0ecef51b1b494e5ed6)